### DISCUSSION TITLE
自製電子書

### DISCUSSION THREAD

**Discussion by eason8479** [1101-GS4719-C/Python-Community-GS4719-C#852] [優秀Python套件]
話說隨著時間過去，漸漸地周遭有許多人都開始用電子書了
而我也加入了這個行列，畢竟帶電子書真的太方便了

不過，有些書要嘛沒有電子版，不然就是很難找
再不然就是載盜版會讓人良心不安

於是我找到了一款app，不過後來發現他是大陸的app
還有許多的資安問題(426不意外)

你剛剛是不是以為我要業配了?
錯了!業配現在才要開始!(誤

好啦!不鬧了，說回正題
Python 的第三方函式庫 PIL 中的 Image 
可以做到一些對圖片的編輯，處理，歸檔等

以下是我藉著前人的努力設計出的小程式

```
from PIL import Image

print('where is the folder location')
url = str(input())

print("how much pic need to convert:")
pic_len = int(input())


image_1 = Image.open(url+"1.jpg")
im_1 = image_1.convert('RGB')
image = [im_1]*pic_len

for i in range (1,pic_len):
    image_tmp = Image.open(url+str(i+1)+'.jpg')
    image[i] = image_tmp.convert('RGB')
  
im_1.save(url+'1.pdf', save_all=True, append_images=image)
```

簡單來說，開啟檔案，讀取檔案，再將其整合成一個PDF
簡單幾行程式碼，一個快速、免費、免下載、免安裝、免聯網、又安全的轉換器就好了
(主要是我不知道怎樣偷資料)

最後再說一點，要用這個還是要有照片源
可以用手機內建的即可
我自己是用Photo scan
當然你要直上專業級設備，解析度拉滿(1張幾GB的那種)我也沒意見啦!

**Comment by dabbie109303501** [comment]
原來這麼簡單就能自製ㄌ

**Comment by Pietra1226** [comment]
這樣就可以減少書包的重量了，python真的很方便

**Comment by Kotori0501** [comment]
這個技術很方便耶，但我從圖片轉pdf都是用手機直接轉的😅

**Comment by dinosaur1009** [comment]
這個功能好方便....不然真的電子書常常有缺漏的部分很麻煩

**Comment by Chiway1229** [comment]
現在有良好轉檔方法了，只差一台平板 ...

**Comment by ching020721** [comment]
電子書真的方便，畢竟原文書很重，下次找不到PDF檔，就來測試看看謝謝分享~

**Comment by StevenWanglolz** [comment]
好厲害~竟然自己寫程式來轉!
我都直接google "image to pdf"，有時候這樣在線上轉檔也滿便捷的!

**Comment by williamyen0623** [comment]
沒想到平常接觸的電子書也能用這種方式自己製作，太酷了吧！～

**Comment by CPCpyt** [comment]
我覺得有震撼到我，或許會改變出版社的門檻和結構。
